src/function.rs

```diff
@@ -372,6 +375,30 @@ fn path_exists(evaluator: &Evaluator, path: &str) -> Result<String, String> {
   )
 }
 
+fn pick(_evaluator: &Evaluator, count: &str, alphabet: Option<&str>) -> Result<String, String> {
+  let alphabet = alphabet.unwrap_or("0123456789abcdef");
+  if alphabet.is_empty() {
+    return Err(String::from("no characters in alphabet"));
+  }
+  let mut alphabet_unique = HashSet::<char>::with_capacity(alphabet.len());
+  for c in alphabet.chars() {
+    if !alphabet_unique.insert(c) {
+      return Err(format!("alphabet contains repeated character {c}"));
+    }
+  }
+  let alphabet = alphabet_unique.drain().collect::<Vec<char>>();
+  let len: usize = match count.parse::<usize>() {
+    Ok(l) => l,
+    _ => return Err(format!("Could not parse `{count}` as a positive integer")),
```

Comment:
```suggestion
    Err(e) => return Err(format!("failed to parse `{count}` as a positive integer: {e}")),
```

Super glad you have the count that was parsed here -- messages like this are missing the thing that failed to parse way too often!